react-app/src/components/SettingsModal/index.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ActionIcon from 'components/ActionIcon';
+import BaseModal from 'components/BaseModal';
+import Logo from 'components/Logo';
+import RadioButton from 'components/RadioButton';
+import SettingsOptionLabel from 'components/SettingsOptionLabel';
+
+import styles from './index.module.scss';
+
+export default class SettingsModal extends React.PureComponent {
+  static propTypes = {
+    /**
+     * The username to display
+     */
+    username: PropTypes.string.isRequired,
+
+    /**
+     * The function to call when hide icon is tapped
+     */
+    onHide: PropTypes.func.isRequired,
+
+    /**
+     * The function to call when logout option is tapped
+     */
+    onSignOut: PropTypes.func.isRequired,
+  }
+
+  state = {
+    visible: false,
+    daily: true,
+  };
+
+  _setDailyNotifications = () => this.setState({ daily: true });
+  _setWeeklyNotifications = () => this.setState({ daily: false });
+
+  _handleHide = () => {
+    this.setState({ visible: false });
+    this.props.onHide(this.state.daily);
+  }
+
+  _renderHeader() {
+    return (
+      <div className={styles.headerContainer}>
+        <div className={styles.hideIcon}>
+          <ActionIcon
+            icon={ActionIcon.HIDE}
+            onClick={this._handleHide}
+          />
+        </div>
+        <div className={styles.usernameContainer}>
+          <div className={styles.calendarIcon} />
+          <div className={styles.username}>
+            {this.props.username}
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  _renderMain() {
+    const { daily } = this.state;
+
+    const notifications = <SettingsOptionLabel
+      icon={SettingsOptionLabel.NOTIFICATION}
+      text={'Notifications'}
+    />
+
+    const signOut = <SettingsOptionLabel
+      icon={SettingsOptionLabel.SIGNOUT}
+      text={'Sign out'}
+      onClick={this.props.onSignOut}
+    />
+
+    const radioButtonDaily = <RadioButton
+      label={"Daily"}
+      selected={daily}
+      onClick={this._setDailyNotifications}
+    />
+
+    const radioButtonWeekly = <RadioButton
+      label={"Weekly"}
```

Comment:
There's no need to use curly braces here.

Reply:
You're right. I'll correct this.